File: python/jwt/security/jwt-none-alg.py
import jwt

def bad1():
    # ruleid: jwt-python-none-alg
    encoded = jwt.encode({'some': 'payload'}, None, algorithm='none')
    return encoded

def bad2(encoded):
    # ruleid: jwt-python-none-alg
    jwt.decode(encoded, None, algorithms=['none'])
    return encoded

def ok(secret_key):
    # ok: jwt-python-none-alg
    encoded = jwt.encode({'some': 'payload'}, secret_key, algorithm='HS256')
    return encoded

File: ruby/jwt/security/jwt-none-alg.rb
require 'jwt'

def bad1
    payload = { data: 'test' }
    # ruleid: ruby-jwt-none-alg
    token = JWT.encode payload, nil, 'none'
    puts token
end

def ok1(hmac_secret)
    # ok: ruby-jwt-none-alg
    token = JWT.encode payload, hmac_secret, 'HS256'
    puts token
    decoded_token = JWT.decode token, hmac_secret, true, { algorithm: 'HS256' }
    puts decoded_token
end
